### PR TITLE
handover: Read timestamp from chain storage instead of System

### PR DESCRIPTION
**Pull request by kvinwang**
This is a fixup to the PR [Safe mode](https://github.com/Phala-Network/phala-blockchain/pull/1157)
Because the info in System would never be updated in safe mode.

### CODE REVIEW

File: crates/phactory/src/prpc_service.rs [hunk 1]
```diff
@@ -1127,8 +1138,9 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> PhactoryApi for Rpc
         _request: (),
     ) -> RpcResult<pb::HandoverChallenge> {
         let mut phactory = self.lock_phactory();
+        let (block, ts) = phactory.current_block()?;
```
Comment (shelvenzhou) [inline crates/phactory/src/prpc_service.rs:1141]: Add comments here about why not use `system.now_ms` and `system.block_number`?

File: crates/phactory/src/prpc_service.rs [hunk 2]
```diff
@@ -1140,14 +1152,14 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> PhactoryApi for Rpc
         let attestation_provider = phactory.attestation_provider;
         let dev_mode = phactory.dev_mode;
         let in_sgx = attestation_provider == Some(AttestationProvider::Ias);
+        let (block_number, now_ms) = phactory.current_block()?;
```
Comment (shelvenzhou) [inline crates/phactory/src/prpc_service.rs:1155]: same as above

File: crates/phactory/src/prpc_service.rs (hunk 1 above)
Comment (kvinwang) [reply crates/phactory/src/prpc_service.rs:1141]: Added alert comments on the fields themselves.

File: crates/phactory/src/prpc_service.rs (hunk 2 above)
Comment (kvinwang) [reply crates/phactory/src/prpc_service.rs:1155]: ditto

**Review by shelvenzhou — APPROVED** [review]
LGTM

**Pull request merged**